src/frontend/staff-portal/src/app/components/jj-dispute-info/jj-count/jj-count.component.ts

```diff
@@ -1,381 +1,445 @@
+        } else {
+          this.form.get('jjDisputedCountRoP').get('stayOfProceedingsBy').disable();
+        }
+
+        // Other fields
+        if (this.jjDisputedCount?.jjDisputedCountRoP?.other) {
+          this.form.get('jjDisputedCountRoP').get('otherCheckbox').setValue(true);
+        } else {
+          this.form.get('jjDisputedCountRoP').get('other').disable();
+        }
+
+        // Latest Plea
+        if (this.jjDisputedCount?.latestPlea) {
+          this.bindLatestPlea(this.jjDisputedCount.latestPlea);
+        }
+
+        // Latest Plea UpdatedTs
+        if (this.jjDisputedCount?.latestPleaUpdateTs) {
+          this.bindLatestPleaUpdateTs(this.jjDisputedCount.latestPleaUpdateTs);
+        }
+      }
+
+      if (this.isViewOnly || !this.jjDisputedCount) {
+        this.form.disable();
+      }
+      
+      this.countForm.patchValue(this.jjDisputedCount);
+        
+      if (this.jjDisputedCount.jjDisputedCountRoP) {
+        this.countRoPForm.patchValue(this.jjDisputedCount.jjDisputedCountRoP);
+        this.countRoPForm.patchValue({
+          _jailIntermittent: this.jjDisputedCount.jjDisputedCountRoP.jailIntermittent === this.JailIntermittent.Y,
+          _dismissed: this.jjDisputedCount.jjDisputedCountRoP.dismissed === this.Dismissed.Y,
+          _forWantOfProsecution: this.jjDisputedCount.jjDisputedCountRoP.forWantOfProsecution === this.ForWantOfProsecution.Y,
+          _withdrawn: this.jjDisputedCount.jjDisputedCountRoP.withdrawn === this.Withdrawn.Y,
+          _abatement: this.jjDisputedCount.jjDisputedCountRoP.abatement === this.Abatement.Y
+        });
+      }
+    }
+  }
+
+  private initListeners() {
+    if (this.jjDisputedCount) {
+      // listen for form changes
+      this.form.valueChanges.subscribe(() => {
+        Object.assign(this.jjDisputedCount, this.form.getRawValue()); // get raw value includes disabled fields
+        if (this.jjDisputedCount.latestPleaUpdateTs) {
+          this.jjDisputedCount.latestPleaUpdateTs = new Date(this.jjDisputedCount.latestPleaUpdateTs).toISOString();
+        }
+        this.jjDisputedCount.includesSurcharge = (this.inclSurcharge === "yes" ? this.IncludesSurcharge.Y : this.IncludesSurcharge.N);
+        this.jjDisputedCountUpdate.emit(this.jjDisputedCount);
+      });
+
+      this.countForm.valueChanges.subscribe(() => {
+        this.jjDisputedCount = { ...this.jjDisputedCount, ...this.countForm.value };
+        this.inclSurcharge = (this.jjDisputedCount.includesSurcharge === this.IncludesSurcharge.Y ? "yes" : "no");
+        this.jjDisputedCountUpdate.emit(this.jjDisputedCount);
+      });
+
+      this.countRoPForm.valueChanges.subscribe(() => {
+        this.jjDisputedCount.jjDisputedCountRoP = { ...this.jjDisputedCount.jjDisputedCountRoP, ...this.countRoPForm.value };
+        this.jjDisputedCount.jjDisputedCountRoP.jailIntermittent = this.countRoPForm.get("_jailIntermittent").value ? this.JailIntermittent.Y : this.JailIntermittent.N;

```

Comment:
this.countRoPForm.value._jailIntermittent

Reply:
Yes, that's cleaner, thanks. Done.